src/Clock.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+class Clock extends React.Component {
+  state = {
+    date: new Date(),
+    name: 0,
+    visibility: false,
+  };
+
+  componentDidMount() {
+    setInterval(() => {
+      this.setState({ date: new Date() });
+    }, 1000);
+  }
```

Comment:
save the result of setInterval into the variable and use in ComponentWillUnmount to clear the interval
https://reactjs.org/docs/state-and-lifecycle.html#adding-lifecycle-methods-to-a-class